File: LightSwitch in Visual Studio 2012 Vision Clinic Walkthrough & Sample/[C#]-LightSwitch in Visual Studio 2012 Vision Clinic Walkthrough & Sample/C#/Vision Clinic CSharp/Common/UserCode/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.LightSwitch;
namespace LightSwitchApplication
{
    public partial class Product
    {
        partial void CurrentPrice_Compute(ref decimal result)
        {
            decimal rebates = default(decimal);

            foreach (var item in ProductRebates)
            {
                if (item.RebateStart <= System.DateTime.Today && item.RebateEnd >= System.DateTime.Today)
                {
                rebates = rebates += item.Rebate.Value;
            }
        }
            result = this.MSRP - rebates;
        }
    }
}

File: LightSwitch in Visual Studio 2012 Vision Clinic Walkthrough & Sample/[VB]-LightSwitch in Visual Studio 2012 Vision Clinic Walkthrough & Sample/VB/Vision Clinic/Common/UserCode/Product.vb

Namespace LightSwitchApplication

    Public Class Product

        Private Sub CurrentPrice_Compute(ByRef result As Decimal)
            ' Set result to the desired field value
            Dim rebates As Decimal
            For Each item In ProductRebates
                If item.RebateStart <= Date.Today And item.RebateEnd >= Date.Today Then
                    rebates = rebates + If(item.Rebate, 0)
                End If
            Next

            result = Me.MSRP - rebates
        End Sub
    End Class

End Namespace
